### PR TITLE
refactor(container-utils): Remove DeltaManagerProxyBase

### PR DESCRIPTION
**Pull request by Josmithr**
DeltaManagerProxyBase was previously deprecated in #16813. It has no external usages. This PR removes it.

### CODE REVIEW

**Review by ghoshkaj — APPROVED** [review]

File: .changeset/sour-jobs-count.md
Comment (tylerbutler) [inline .changeset/sour-jobs-count.md:8]: Can you include the minor version this was deprecated in? It really helps partners. I know we're confident there are no usages of this, but we need to get into the habit of including these details to help partners trace changes.

**Review by tylerbutler — APPROVED** [review]
Exciting to see all the removals! I had a note about the changeset but looks good otherwise.

File: .changeset/sour-jobs-count.md
Comment (tylerbutler) [inline .changeset/sour-jobs-count.md:8]:
```suggestion
`DeltaManagerProxyBase` was deprecated in [version 2.0.0-internal.6.1.0](https://github.com/microsoft/FluidFramework/releases/tag/client_v2.0.0-internal.6.1.0). It is no longer exported, and no replacement API is intended.
```

Comment (Josmithr) [reply .changeset/sour-jobs-count.md]: Yeah, that's a good call. Will update.

Comment (Josmithr) [reply .changeset/sour-jobs-count.md]: JK. You already got it. Github's view is so inconsistent... Thank you!!

**Comment by tylerbutler** [comment]
I'll merge this once #16941 is merged.

**Pull request merged**
